assets/wizards/readerRevenue/views/stripe-setup/index.js

```diff
@@ -103,6 +105,115 @@ export const StripeKeysSettings = () => {
 	);
 };
 
+const WebhooksList = ( { webhooks, onUpdate } ) => {
+	const { wizardApiFetch } = useDispatch( Wizard.STORE_NAMESPACE );
+	const isLoading = useSelect( select => select( Wizard.STORE_NAMESPACE ).isQuietLoading() );
+
+	const handleChange = ( webhook, method, payload ) => {
+		const args = {
+			path: `/newspack/v1/stripe/webhook/${ webhook.id }`,
+			method,
+			data: payload,
+			isQuietFetch: true,
+		};
+		if ( payload ) {
+			args.data = payload;
+		}
+		wizardApiFetch( args ).then( onUpdate( method ) );
+	};
+
+	return (
+		<ul>
+			{ webhooks.map( webhook => {
+				return (
+					<li key={ webhook.id }>
+						<span>
+							<ToggleControl
+								checked={ webhook.status === 'enabled' }
+								disabled={ isLoading }
+								onChange={ () =>
+									handleChange( webhook, 'POST', {
+										status: webhook.status === 'enabled' ? 'disabled' : 'enabled',
+									} )
+								}
+							/>
+							<code>{ webhook.url }</code>
+						</span>
+
+						<Button
+							isDestructive
+							icon={ trash }
+							disabled={ isLoading }
+							onClick={ () => handleChange( webhook, 'DELETE' ) }
+						/>
+					</li>
+				);
+			} ) }
+		</ul>
+	);
+};
+
+const StripeWebhooksSettings = () => {
+	const { stripe_data: data = {} } = Wizard.useWizardData( 'reader-revenue' );
+	const { updateWizardSettings } = useDispatch( Wizard.STORE_NAMESPACE );
+	const webhooksList = data.webhooks_list || [];
+
+	const [ thisSiteWebhooks, otherWebhooks ] = webhooksList.reduce(
+		( acc, webhook ) => {
+			if ( webhook.matches_url ) {
+				acc[ 0 ].push( webhook );
+			} else {
+				acc[ 1 ].push( webhook );
+			}
+			return acc;
+		},
+		[ [], [] ]
+	);
+	const activeThisSiteWebhooks = thisSiteWebhooks.filter( webhook => webhook.status === 'enabled' );
+
+	const handleWebhooksListUpdate = method => payload => {
+		let newList = webhooksList;
+		switch ( method ) {
+			case 'POST':
+				newList = webhooksList.map( webhook => ( payload.id === webhook.id ? payload : webhook ) );
+				break;
+			case 'DELETE':
+				newList = webhooksList.filter( webhook => payload.id !== webhook.id );
+				break;
+		}
+		updateWizardSettings( {
+			slug: READER_REVENUE_WIZARD_SLUG,
+			path: [ 'stripe_data', 'webhooks_list' ],
+			value: newList,
+		} );
+	};
+	return (
+		<div className="newspack-payment-setup-screen__webhooks">
+			{ activeThisSiteWebhooks.length > 1 && (
+				<Notice
+					isError
+					noticeText={ __(
+						'There are too many webhoooks for this site. Please delete or disable the extra ones.',
+						'newspack'
+					) }
+				/>
+			) }
+			{ activeThisSiteWebhooks.length === 0 && (
+				<Notice
+					isError
+					noticeText={ __( 'There are no active webhoooks for this site.', 'newspack' ) }
```

Comment:
It's possible (in fact, very easy) to delete the active webhook for your site. Once that happens it seems like there's nothing to be done to bring it back, but if you refresh the page it's recreated automatically. Perhaps we could show a.) a warning before triggering the deletion, and b.) a "reset" button after deletion that will recreate the webhook and update the UI in this instance. This could avoid some panic that might occur if someone clicks the delete button by accident. In general I can also see a reset button being useful in case a webhook goes haywire on the Stripe end—you may want to recreate it to fix things.

Reply:
@thomasguillot could use your opinion on this, too. Maybe I'm overthinking things.

Reply:
I don't really understand the "Reset Webhook" thing but if you think it makes sense then we could add the button to the list of buttons at the bottom (see https://github.com/Automattic/newspack-plugin/pull/2143#discussion_r1040915445)

Reply:
Maybe I'm overcomplicating it. My main concern is that it seems too easy to accidentally delete the active webhook for the current site. Maybe the only change needed is to change the "trash" button on that one webhook to a "reset" button that will delete and recreate the webhook and update the UI accordingly. Or, more simply, to just remove the trash button entirely if we don't think that's a use case we need to support.

Reply:
I think a reset button, and a confirmation dialog before webhook deletion and reset will solve these issues.

Reply:
> I think a reset button, and a confirmation dialog before webhook deletion and reset will solve these issues.

I'm happy with this and moving the webhooks to a new sub-page as discussed above. Thanks for the discussion, @adekbadek and @thomasguillot!

Reply:
Discussion continued in the [other thread](https://github.com/Automattic/newspack-plugin/pull/2143#discussion_r1036567641).